python/valuecell/agents/common/trading/_internal/coordinator.py

```diff
@@ -132,7 +133,11 @@ async def run_once(self) -> DecisionCycleResult:
                     portfolio.buying_power = float(free_cash)
                     # Also update free_cash field in view if it exists
                     portfolio.free_cash = float(free_cash)
-
+                positions = await fetch_positions_from_gateway(self._execution_gateway)
+                if len(positions) == 0:
```

Comment:
Skipping the sync when `len(positions) == 0` introduces a bug. If the positions are fully liquidated (e.g., manually closed), this check prevents portfolio.positions from being cleared, leaving stale positions in memory.

Additionally, this logic makes it impossible to distinguish between a legitimately empty portfolio and a data fetch failure (if failure results in an empty list).